### PR TITLE
feat: make OV login as soon as challenge is done

### PR DESCRIPTION
**Pull request by ivandenysenko-okta**
## Description:

Log in as soon as challenge request is finished.

## PR Checklist

- [x] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Screenshot/Video:

ezgif-3-fd26a26aada0

### Issue:

- [OKTA-367901](https://oktainc.atlassian.net/browse/OKTA-367901)

### CODE REVIEW

File: test/testcafe/spec/DeviceChallengePollViewFailure_spec.js [hunk 1]
```diff
@@ -36,7 +36,6 @@ async function setup(t) {
 test('probing and polling APIs are sent and responded', async t => {
   const deviceChallengePollPageObject = await setup(t);
   await t.expect(deviceChallengePollPageObject.getHeader()).eql('Verifying your identity');
-  await t.expect(deviceChallengePollPageObject.getSpinner().getStyleProperty('display')).eql('block');
```
Comment (ivandenysenko-okta) [inline test/testcafe/spec/DeviceChallengePollViewFailure_spec.js:39]: Now when we get probing response, log in happens immediately, so no spinner is show.

File: src/v2/view-builder/views/DeviceChallengePollView.js [hunk 2]
```diff
@@ -148,7 +148,10 @@ const Body = BaseForm.extend(Object.assign(
           // TODO: can we use standard ES6 promise methods, then/catch?
           .done(() => {
             this.probingXhr = onPortFound();
-            return this.probingXhr;
+            return this.probingXhr.done(() => {
+              // log in as soon as challenge request is finished
+              return this.options.appState.trigger('saveForm', this.model);
```
Comment (haishengwu-okta) [inline src/v2/view-builder/views/DeviceChallengePollView.js:153]: shall be `this.trigger('save', this.model)`

**Review by haishengwu-okta — APPROVED** [review]
- one comment

File: src/v2/view-builder/views/DeviceChallengePollView.js (hunk 2 above)
Comment (ivandenysenko-okta) [reply src/v2/view-builder/views/DeviceChallengePollView.js]: done ✔️

**Pull request closed without merging**